File: Data Structures & Algorithms/buildings-with-an-ocean-view/submission-0.cpp
class Solution {
public:
    vector<int> findBuildings(vector<int>& heights) {
    int maxl=0;
    vector<int> ans;
    for (int i=heights.size()-1;i>=0;i--)
    {
        if (heights[i]>maxl)
         ans.push_back(i);
        maxl=max(maxl,heights[i]);
    }
    reverse(ans.begin(),ans.end());
    return ans;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l=0
        st=set()
        maxl=0
        for right in range(len(s)):
            while s[right] in st:
                st.remove(s[l])
                l+=1
            st.add(s[right])
            maxl=max(maxl,right-l+1)
        return maxl

